### ISSUE TITLE
Calling `importance` with different choicemap shapes can be very expensive in JIT time

### ISSUE THREAD

**Issue by georgematheos** [genjax-community/genjax#1463]
I have a large generative function, where each GFI method takes about 2 minutes to run the first time due to JIT compilation.

In my test cases, I generate lots of test data out of the generative function.  I have ~5 different modes in which I do constrained generation from my generative function, constraining different sets of addresses, to generate traces with different properties.  Each of these modes involves calling `model.importance` with a choicemap with a different shape.  Each of these importance calls takes about 2 minute to run, for a total of 10 minutes.

I wish there were a way that I could only pay the cost of JIT compiling `model.importance` once, and have the resulting JITted importance function be applicable to all these different choicemap shapes.  This issue proposes to find a way to allow users to do this.

One approach might be to provide users with utility functions that let them say "given my choicemap with addresses (a, b, c), turn this into a masked choicemap that has addresses (a, b, c, d, e, f) but has masked out everything except addresses (a, b, c)".  Then, I could look at my testing code, and say "okay, between in each test where I call `importance`, I have a choicemap with some subset of the addresses in (a, b, c, d, e, f).  So, at the top of my testing file, I'll write down a selection containing all of (a, b, c, d, e, f), and then before each `importance` call with a choicemap containing a subset of these addresses, I will use GenJAX's util to lift that smaller choicemap to a choicemap masked choicemap that has the maximal set of possible addresses in its type signature".

This util could look something like this:
```python
all_addresses = S["a"] | S["b"] | S["c"] | S["d"] | S["e"] | S["f", S.Idx, "g"]
my_chm_1 = C.kw(a=1, b=2, c=3)
lifted_chm_1 = genjax.lift_choicemap(my_chm_1, all_addresses)
my_chm_2 = C.kw(c=5, e=1, f=C[2, "g"].set(1.0))
lifted_chm_2 = genjax.lift_choicemap(my_chm_2, all_addresses)

model.importance(key, lifted_chm_1, args) # JITs
model.importance(key, lifted_chm_2, args) # no need to JIT,
                         # since the pytree shape for `lifted_chm_2` is the same
                         # as that for `lifted_chm_1`
```

**Comment by arijit-dasgupta** [comment]
I love this idea! Although one needs to know perhaps the hierarchy of addresses for more hierarchical/complex choicemaps, so it might be easier to give it the selection via `chm.get_selection()` if you already have the choicemap structure in place, or find some other way to specify it. @georgematheos you can manually construct switch choicemaps via `ChoiceMap.switch(jnp.int_(0), [chm_1, chm_2,...])`. This way, your importance will not re-JIT. This [issue](https://github.com/probcomp/genjax/issues/1455) tells me you may have already been using this for a while. I do like the idea of automating the generation of `chm_1`, `chm_2`,.., too. Although my use case is different, where I am using it as a proposal, so my `chm_1`, `chm_2` is automatically created via a switch combinator, and I do a simple choicemap translation. There is a bug + workaround for this in the case where you want to constrain on both switches and non-switches in the same GFI call. You can see my issue [here](https://github.com/probcomp/genjax/issues/1466).
